pkg/test/utils/kubernetes.go

```diff
@@ -1321,3 +1398,29 @@ func Kubeconfig(cfg *rest.Config, w io.Writer) error {
 		},
 	}, w)
 }
+
+func constructGVK(apiVersion, kind string) schema.GroupVersionKind {
+	apiVersionSplit := strings.Split(apiVersion, "/")
+	gvk := schema.GroupVersionKind{
+		Version: apiVersionSplit[len(apiVersionSplit)-1],
+		Kind:    kind,
+	}
+	if len(apiVersion) > 1 {
+		gvk.Group = apiVersionSplit[0]
+	}
+
+	return gvk
+}
+
+func NewClient(kubeconfig, context string) func(bool) (client.Client, error) {
```

Comment:
This file is already very long as is. Can we perhaps move this (maybe along with `NewRetryClient`) into the `.../pkg/k8s` package?

Reply:
Good point but that creates an import cycle. :(

```shell
package command-line-arguments
        imports github.com/kudobuilder/kuttl/pkg/kuttlctl/cmd
        imports github.com/kudobuilder/kuttl/pkg/k8s
        imports github.com/kudobuilder/kuttl/pkg/test/utils
        imports github.com/kudobuilder/kuttl/pkg/k8s: import cycle not allowed
```

Reply:
Maybe we could raise a separate PR to refactor the file?

Reply:
Good idea. Without looking at details, I think `pkg/k8s` should be the leaf 🤔

Reply:
> I think pkg/k8s should be the leaf?

Yes, that makes sense.

Reply:
Now that this function is again only called from `(t *Case) Run()` it can go back to being an unexported `newClient` in `pkg/test/case.go`, right?